gcc/rust/typecheck/rust-hir-type-check-type.cc

```diff
@@ -395,10 +395,17 @@ TypeCheckType::resolve_root_path (HIR::TypePath &path, size_t *offset,
 	  auto nr_ctx
 	    = Resolver2_0::ImmutableNameResolutionContext::get ().resolver ();
 	  // assign the ref_node_id if we've found something
-	  nr_ctx.lookup (path.get_mappings ().get_nodeid ())
-	    .map ([&ref_node_id, &path] (NodeId resolved) {
-	      ref_node_id = resolved;
-	    });
+	  nr_ctx.lookup (ast_node_id).map ([&ref_node_id] (NodeId resolved) {
```

Comment:
you need to be looking at the segment node_id.

The name resolution for Paths is a really huge pain in the ass. this is resolving the _ROOT_ of a path so for example you could have:

mod_a::Type::impl_item

the root path should just resolve this:

mod_a::Foo then stop then fall back to the resolve segments to then find the impl_item

So i think you should just be using the ast_node_id like i do in the old name resolver here.